service/history/mutableStateBuilder.go

```diff
@@ -3939,6 +3945,21 @@ func (e *mutableStateBuilder) CloseTransactionAsSnapshot(
 	return workflowSnapshot, workflowEventsSeq, nil
 }
 
+func (e *mutableStateBuilder) IsEventDuplicated(
+	event definition.DeduplicationKey,
```

Comment:
event is kind of misleading, maybe resourceDedupKey?

Reply:
done